### ISSUE TITLE
feat: pytree PoC for parametric lp FitImaging (lp_pytree)

### ISSUE THREAD

**Issue by Jammy2211** [Jammy2211/PyAutoLens#450]
## Overview

Extend Path A (`jax.jit`-wrapped `analysis.fit_from`) to a parametric light-profile source variant
(`Sersic`, no MGE, no pixelization, no linear-light-profile), so that `jax.jit(analysis.fit_from)(instance)`
returns a fully-populated `FitImaging` whose `log_likelihood` matches the NumPy path.

This is a cheap regression test on top of the MGE PoC (already shipped). The prompt expects
"probably nothing new" — the deliverable is locking in existing behaviour for the simpler
single-profile parametric path.

## Plan

- Scaffold `autolens_workspace_test/scripts/jax_likelihood_functions/imaging/lp_pytree.py` from
  the `mge_pytree.py` template.
- Use the `lp.py` model topology (lens: `Sersic` bulge + `PowerLaw` mass + `ExternalShear`;
  source: `Sersic` bulge), but switch from `lp_linear.Sersic` to `lp.Sersic` per the prompt's
  explicit intent (parametric, not linear).
- Run, iterate registrations only if anything surfaces.
- Confirm `PASS: jit(fit_from) round-trip matches NumPy scalar.`

<details>
<summary>Detailed implementation plan</summary>

### Affected Repositories
- autolens_workspace_test (primary — deliverable)
- PyAutoLens, PyAutoGalaxy, PyAutoArray, PyAutoFit (attached for any registration fallout)

### Work Classification
Both — workspace test script primary, library registrations possible

### Branch Survey

| Repository | Current Branch | Dirty? |
|-----------|---------------|--------|
| autolens_workspace_test | main | clean |
| PyAutoLens | main | clean |
| PyAutoGalaxy | main | clean |
| PyAutoArray | main | clean |
| PyAutoFit | main | clean |

**Suggested branch:** `feature/fit-imaging-pytree-lp`
**Worktree root:** `~/Code/PyAutoLabs-wt/fit-imaging-pytree-lp/` (created next by `/start_library`)

### Implementation Steps

1. Copy `mge_pytree.py` template to `lp_pytree.py`.
2. Splice in dataset/mask setup from `lp.py` (mask radius 3.0, no radial-bin oversampling).
3. Build the model with `al.lp.Sersic` for both lens bulge and source bulge (per prompt intent).
4. Mass: `al.mp.PowerLaw`, shear: `al.mp.ExternalShear`.
5. Register model with `register_model(model)`.
6. Compute NumPy reference via `al.AnalysisImaging(..., use_jax=False)`.
7. JIT-wrap `analysis.fit_from`, assert `log_likelihood` matches.

### Key Files
- `autolens_workspace_test/scripts/jax_likelihood_functions/imaging/lp_pytree.py` (NEW)
- Library: any registration fallout → `PyAutoLens/autolens/imaging/model/analysis.py::_register_fit_imaging_pytrees`

</details>

## Original Prompt

<details>
<summary>Click to expand starting prompt</summary>

```
Extend Path A (jax.jit-wrapped \`analysis.fit_from\`) to @PyAutoLens/autolens/imaging/fit_imaging.py
\`FitImaging\` with a **parametric light-profile** source (Sersic / Exponential etc., no MGE, no
pixelization), so that \`jax.jit(analysis.fit_from)(instance)\` returns a fully-populated \`FitImaging\`
with \`jax.Array\` leaves whose \`log_likelihood\` matches the NumPy path.
```

(Full prompt at admin_jammy/prompt/issued/fit_imaging_pytree_lp.md after move.)

</details>

**Comment by Jammy2211** [comment]
## Shipped — 2026-04-19

### PRs
- https://github.com/PyAutoLabs/autolens_workspace_test/pull/35

### Summary
Added `lp_pytree.py` as a sibling to `mge_pytree.py` covering the single-profile parametric
(non-linear) light-profile path. `jax.jit(analysis.fit_from)` round-trip matches the NumPy
scalar exactly (-190300716.097) with **no new pytree registrations** required. This confirms
the prompt's expectation that parametric single-profile is strictly simpler than the MGE
basis path already shipped.

### Session Notes
- Prompt and reference script disagreed on linear vs non-linear: `lp.py` uses
  `al.lp_linear.Sersic` despite the prompt saying "parametric…does not use linear light
  profiles". Honoured the prompt's explicit intent — built the PoC with `al.lp.Sersic`,
  which tests a distinct code path from the MGE/linear ship.
- PASS on first iteration. Cheap regression-locking ship, no library changes.

**Comment by liuyuzhe530** [comment]
看了一下问题，给出以下排查建议：

1. 检查官方文档和changelog
2. 确认依赖版本兼容性
3. 提供完整的错误日志

需要付费协助请联系我：
📧 liuyuzhe530@gmail.com
🌐 https://github.com/liuyuzhe530

**Comment by Jammy2211** [comment]
Shipped via PyAutoLabs/autolens_workspace_test#35. Closing.
